.github/workflows/test.yml

```diff
@@ -49,7 +49,7 @@ jobs:
 
     - if: matrix.python-version == '3.8' && matrix.os == 'ubuntu-latest'
       name: Check Style
-      run: poetry run poe check_style
+      run: poetry run pre-commit run --all-file --verbose
```

Comment:
Should probably use `uses: pre-commit/action@v3.0.0` otherwise pre-commit does not cache the tools it has to install. Installing the tools takes longer than running all the checks!

Should probably still keep pre-commit in `pyproject.toml` so that contributors don't need to manually install pre-commit.

Reply:
switched to github action and also bumped the version of the other github actions.